web/packages/extension/webpack.config.js

```diff
@@ -55,11 +55,24 @@ function transformManifest(content, env) {
                 id: firefoxExtensionId,
             },
         };
+
+        // Firefox does not currently support background service worker,
+        // see https://bugzilla.mozilla.org/show_bug.cgi?id=1573659
+        manifest.background = { scripts: ["dist/background.js"] };
+        // Firefox supports declarativeNetRequest starting with version 113,
+        // but it's too soon to update to a Firefox version that recent
+        // Firefox supposedly supports the scripting API, but not the MAIN
+        // execution world (https://bugzilla.mozilla.org/show_bug.cgi?id=1736575)
+        manifest.permissions = ["storage", "webRequest", "webRequestBlocking"];
```

Comment:
Does firefox now support `host_permissions` for `webRequest`, or do we need to readd `<all_urls>` here?

Reply:
As of Firefox 109 it's supported: <https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/host_permissions#browser_compatibility>.

I didn't realize this would up the minimum supported Firefox version that much anyway though.

Reply:
Apparently that's when MV3 released for Firefox anyway: <https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/109#changes_for_add-on_developers>